src/xinterpreter_raw.cpp

```diff
@@ -0,0 +1,348 @@
+                    m_displayhook.attr("set_execution_count")(execution_count);
+                }
+
+                exec(compiled_code);
+                exec(compiled_interactive_code);
+            }
+            else
+            {
+                py::object compiled_code = builtins.attr("compile")(code_ast, filename, "exec");
+                exec(compiled_code);
+            }
+
+            kernel_res["status"] = "ok";
+            kernel_res["user_expressions"] = nl::json::object();
+            kernel_res["payload"] = nl::json::array();
+
+        }
+        catch (py::error_already_set& e)
+        {
+            xerror error = extract_already_set_error(e);
+            
+            if (!silent)
+            {
+                publish_execution_error(error.m_ename, error.m_evalue, error.m_traceback);
+            }
+
+            kernel_res["status"] = "error";
+            kernel_res["ename"] = error.m_ename;
+            kernel_res["evalue"] = error.m_evalue;
+            kernel_res["traceback"] = error.m_traceback;
+        }
+
+        // Cache inputs
+        py::globals()["_iii"] = py::globals()["_ii"];
+        py::globals()["_ii"] = py::globals()["_i"];
+        py::globals()["_i"] = code;
+
+        return kernel_res;
+    }
+
+    nl::json raw_interpreter::complete_request_impl(
+        const std::string& code,
+        int cursor_pos)
+    {
+        py::gil_scoped_acquire acquire;
+        nl::json kernel_res;
+
+        py::list completion = py::list(3);
+
+        kernel_res["matches"] = completion[0];
+        kernel_res["cursor_start"] = completion[1];
+        kernel_res["cursor_end"] = completion[2];
+        kernel_res["metadata"] = nl::json::object();
+        kernel_res["status"] = "ok";
+
+        return kernel_res;
+    }
+
+    nl::json raw_interpreter::inspect_request_impl(const std::string& code,
+        int cursor_pos,
+        int detail_level)
+    {
+        py::gil_scoped_acquire acquire;
+        nl::json kernel_res;
+        nl::json data = nl::json::object();
+        bool found = false;
+
+        py::module tokenutil = py::module::import("IPython.utils.tokenutil");
+        py::str name = tokenutil.attr("token_at_cursor")(code, cursor_pos);
+
+
+        kernel_res["data"] = data;
+        kernel_res["metadata"] = nl::json::object();
+        kernel_res["found"] = found;
+        kernel_res["status"] = "ok";
+        return kernel_res;
+    }
+
+    nl::json raw_interpreter::is_complete_request_impl(const std::string& code)
+    {
+        py::gil_scoped_acquire acquire;
+        nl::json kernel_res;
+
+        py::object transformer_manager;
+        py::list result = transformer_manager.attr("check_complete")(code);
+        auto status = result[0].cast<std::string>();
+
+        kernel_res["status"] = status;
+        if (status.compare("incomplete") == 0)
+        {
+            kernel_res["indent"] = std::string(result[1].cast<std::size_t>(), ' ');
+        }
+        return kernel_res;
```

Comment:
I believe this doesn't work.

Let's not implement this request for now. It will not be used by Voila or JupyterLab anyway, it's only for the console.